diff --git a/jormungandr-lib/src/interfaces/block0_configuration/mod.rs b/jormungandr-lib/src/interfaces/block0_configuration/mod.rs
index c341a385..797ff044 100644
--- a/jormungandr-lib/src/interfaces/block0_configuration/mod.rs
+++ b/jormungandr-lib/src/interfaces/block0_configuration/mod.rs
@@ -132,6 +132,33 @@ mod test {
         }
     }
 
+    #[test]
+    // Can be run with:
+    // $ cd jormungandr-lib
+    // $ cargo test write_arbitrary_block0s_to_disk
+    //
+    // Then the blocks can be moved to cardano-wallet:
+    // $ mv block0-*.bin ../../cardano-wallet/lib/jormungandr/test/data/block0s/
+    // here assuming jormungandr and cardano-wallet live in the same directory.
+    fn write_arbitrary_block0s_to_disk() {
+        use std::fs::File;
+        use chain_core::property::{Serialize};
+        for i in 1..10 {
+            let file = File::create(format!("block0-{:?}.bin", i)).unwrap();
+            let _ = arbitrary_block0().serialize(file);
+        }
+    }
+    fn arbitrary_block0() -> Block {
+        use quickcheck::{StdGen};
+        use rand::{thread_rng};
+        let rng = thread_rng();
+        // We can control the size of the generated values here
+        let mut g = StdGen::new(rng, 1000);
+        let a: Block0Configuration = Arbitrary::arbitrary(&mut g);
+        return a.to_block();
+    }
+
+
     #[test]
     fn documented_example_decodes() {
         let _: Block0Configuration =
